From 83a9c02958db67394767c8383813a47a0a1b8feb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Romain=20Roff=C3=A9?= <r2@kyber.tech>
Date: Fri, 24 Apr 2026 14:22:20 +0200
Subject: [PATCH 2/3] libavutil/d3d11va: Avoid to query ID3D11VideoDevice and
 ID3D11VideoContext

Required to make DXGI work when not physical GPU is available.
---
 libavutil/hwcontext_d3d11va.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/libavutil/hwcontext_d3d11va.c b/libavutil/hwcontext_d3d11va.c
index 418fe9dfda..b3bab1af8b 100644
--- a/libavutil/hwcontext_d3d11va.c
+++ b/libavutil/hwcontext_d3d11va.c
@@ -525,20 +525,6 @@ static int d3d11va_device_init(AVHWDeviceContext *hwdev)
             return AVERROR_UNKNOWN;
     }
 
-    if (!device_hwctx->video_device) {
-        hr = ID3D11DeviceContext_QueryInterface(device_hwctx->device, &IID_ID3D11VideoDevice,
-                                                (void **)&device_hwctx->video_device);
-        if (FAILED(hr))
-            return AVERROR_UNKNOWN;
-    }
-
-    if (!device_hwctx->video_context) {
-        hr = ID3D11DeviceContext_QueryInterface(device_hwctx->device_context, &IID_ID3D11VideoContext,
-                                                (void **)&device_hwctx->video_context);
-        if (FAILED(hr))
-            return AVERROR_UNKNOWN;
-    }
-
     return 0;
 }
 
-- 
2.47.3

